scholarly/_scholarly.py

```diff
@@ -17,6 +17,22 @@ def search_pubs(self, query):
         url = _PUBSEARCH.format(requests.utils.quote(query))
         return self.nav.search_publications(url)
 
+    def search_pub_list(self, publist: list, filled=False):
```

Comment:
I would skip including the `search_pub_list` and `search_terms` methods. They are easy to implement outside the API if someone needs them, and it is better to keep the API simpler, imho.